js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +34,86 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      clicked: 0,
     }
   }
+  /**
+   * Change the clicked and courseId state, whenever a course link is clicked.
+   */
+  clickRelated = s => {
+    let copyState = this.state
+    copyState.courseId = s
+    copyState.clicked = 1
+    this.setState(copyState)
+  }
+
+  /**
+   * Convert the course names to <a> tags and return a list of strings and <a> tags.
+   */
+  getRelatedCourses = content => {
+    const result = []
+    if (content !== null) {
+      content.split(" ").forEach((word, i) => {
+        if (word.match(/[A-Z]{3,4}[0-9]{3}[H|Y]1/gi)) {
```

Comment:
Pull out this regular expression into a constant.

Reply:
Also, about the regex itself:

1. I'm not sure there are any course codes in FAS that have 4 letters, did you find such examples?
2. Make sure this regex matches the entire string, not just a substring. Otherwise it'll result in false positives.